lib/search.rb

```diff
@@ -337,9 +337,27 @@ def execute(readonly_mode: Discourse.readonly_mode?)
 
     find_grouped_results if @results.posts.blank?
 
-    if preloaded_topic_custom_fields.present? && @results.posts.present?
+    if @results.posts.present?
       topics = @results.posts.map(&:topic)
-      Topic.preload_custom_fields(topics, preloaded_topic_custom_fields)
+
+      if preloaded_topic_custom_fields.present?
+        Topic.preload_custom_fields(topics, preloaded_topic_custom_fields)
+      end
+
+      if @guardian.can_lazy_load_categories?
+        topics.each do |topic|
+          if topic.category.present?
+            # Adding categories directly to the internal list to avoid
+            # per_facets / per_limits limits
+            if topic.category.parent_category.present?
+              @results.categories << topic.category.parent_category
```

Comment:
I'm not sure I understand this because adding this to `@results` is saying that the categories matches the given search term when it actually does not.